misc/rapiddisk-rootdev/install_initrd.sh

```diff
@@ -104,137 +143,369 @@ if [ -z "$uninstall" ]; then
 				root_device=/dev/"$(ls 2>/dev/null -l $device | grep -oP '[^/]+$')"
 				;;
 			*)
-				myerror "Error: could not find the root device from /etc/fstab. Use the '--root' option. Exiting..."
+				myerror "could not find the root device from /etc/fstab. Use the '--root' option."
 				;;
 		esac
 
 		# TODO this check must be improved
 		if ! echo "$root_device" | grep -P '^/dev/\w{1,4}\d{0,99}$' >/dev/null 2>/dev/null ; then
-			myerror "Error: root_device '$root_device' must be in the form '/dev/xxx' or '/dev/xxxn with n as a positive integer. Exiting..."
+			myerror "root_device '$root_device' must be in the form '/dev/xxx' or '/dev/xxxn with n as a positive integer."
 		fi
 
 		echo " - Root device '$root_device' was found!"
 		echo ' - Is it ok to use it? [yN]'
-		read yn
+		read -r yn
 
-		{ [ "$yn" = "y" ] || [ "$yn" = "Y" ] ; } || myerror "Error: the root device we found was not ok. Use the '--root' option. Exiting..."
+		{ [ "$yn" = "y" ] || [ "$yn" = "Y" ] ; } || myerror "the root device we found was not ok. Use the '--root' option."
 	fi
 
-	[ -n "$ramdisk_size" ] ||  myerror "Error: missing argument '--size'. Exiting..."
-	is_num "$ramdisk_size" || myerror "Error: The ramdisk size must be a positive integer. Exiting..."
-	
+}
 
-fi
+
+# we check for user == root
+whoami | grep root 2> /dev/null 1> /dev/null || myerror "sorry, this must be run as root."
```

Comment:
what if the username is "groot" and not "root"? The grep will return successfully. We need to refine this and make it something like: whoami |grep "^root$"

Reply:
Ouch! Fixed!